openfifth/aspen-discovery
# Summon usage graphs

**Description**

An Aspen admin can now view a usage graph and a raw data table for each of the variables included in the Summon Dashboard. This feature used to be limited to the System Reports dashboard, is being added to the ILS Integration [see this PR  ](https://github.com/PTFS-Europe/aspen-discovery/pull/91), and it is now being extended further.

Note: as with ILS Integration, the implementation for Summons is based on the the System Reports existing usage graphs, and intends to mirror it where possible

Features:

- The dashboard now includes links next to the title of each of its sections
- Those links lead to the relevant Usage Graph page
- The Usage Graph page contains both a graph and a raw data table
- While the user is on a given Summon Usage Graph page, the Summon section on the left hand-side options menu remains open (active)

**Test plan:**

1. apply the patch
2. log in as an admin
3. navigate to Administration Settings > Modules > Summon, and tick the 'enable' checkbox before saving (this will allow the 'Summon' section to display on the Administration left hand-side navigation bar)
4. ensure that the user_summon_usage and summon_usage table contain some data: if not, insert mock data in both tables
5. navigate to Administration Settings > Summon > Dashboard
6. click on the link displayed next to a section's header (title) and access the relevant Usage Graph page
7. notice that the data displayed matches the summary found on the dashboard
